File: firebase/firestore/node/feedback.dart
// @dart=2.9

import "package:firestore/data.dart";
import "package:firestore/helpers.dart";
import "package:firestore/services.dart";

Future<void> main() async {
	Args.initLogger("Getting feedback");
	Logger.level = LogLevel.debug;

	final List<Feedback> feedback = await Firestore.getFeedback();

	feedback.sort(
		(Feedback a, Feedback b) => a.timestamp.compareTo(b.timestamp)
	);

	for (final Feedback message in feedback) {
		Logger.verbose("\t$message");
	}

	await app.delete();
	Logger.info("Feedback download complete");
}

File: firebase/firestore-py/bin/feedback.py
from firebase_admin import delete_app

import lib.utils as utils
import lib.services as firebase

if __name__ == '__main__':
	utils.logger.info("Getting feedback...")
	feedback = firebase.get_feedback()
	feedback.sort()
	for message in feedback:
		utils.logger.debug("Feedback", str(message))

	delete_app(firebase.app)
	utils.logger.info("Got feedback")